embedded/sql/engine.go

```diff
@@ -555,7 +555,7 @@ func (e *Engine) loadTables(db *Database, catalogSnap, dataSnap *store.Snapshot)
 				return ErrCorruptedData
 			}
 
-			table.maxPK = binary.BigEndian.Uint64(encMaxPK)
+			table.maxPK = int64(binary.BigEndian.Uint64(encMaxPK) + math.MaxInt64 + 1)
```

Comment:
I would go for just plain cast between signed and unsigned value here:

```suggestion
			table.maxPK = int64(binary.BigEndian.Uint64(encMaxPK))
```

It would be backwards-compatible and IIRC such casts are well-defined in golang, and that's how it's done in go's stdlib: https://cs.opensource.google/go/go/+/refs/tags/go1.17.1:src/encoding/binary/binary.go;l=329

Reply:
the shift when mapping from signed to unsigned space is needed in order to provide lexicographic order when encoding (big-endian)  e.g. -3 -1 0 1 3
Otherwise -1 would become the greatest one.

Reply:
I agree on that for the index - we'd have to shift the range for correct sorting.
But I'm not so sure about the payload itself. Receiving proof for a row would not be so obvious anymore - it's not a trivial big-endian integer anymore.

Reply:
The proofs and indexing take place at the kv-level. At that layer we deal just with byte-arrays, so we need to serialize the data in the same way for storing, indexing and to build proofs